feature-libs/product-configurator/rulebased/components/configurator-conflict-and-error-messages/configurator-conflict-and-error-messages.component.spec.ts

```diff
@@ -0,0 +1,269 @@
+import {
+  CommonConfigurator,
+  CommonConfiguratorTestUtilsService,
+  CommonConfiguratorUtilsService,
+} from '@spartacus/product-configurator/common';
+import { IconLoaderService } from '@spartacus/storefront';
+import { Observable, of } from 'rxjs';
+import { ConfiguratorCommonsService } from '../../core/facade/configurator-commons.service';
+import { Configurator } from '../../core/model/configurator.model';
+import { ConfiguratorConflictAndErrorMessagesComponent } from './configurator-conflict-and-error-messages.component';
+
+const PRODUCT_CODE = 'CONF_LAPTOP';
+const CONFIG_ID = '12342';
+const CONFIGURATOR_ROUTE = 'configureCPQCONFIGURATOR';
+
+const mockRouterState: any = {
+  state: {
+    params: {
+      entityKey: PRODUCT_CODE,
+      ownerType: CommonConfigurator.OwnerType.PRODUCT,
+    },
+    semanticRoute: CONFIGURATOR_ROUTE,
+  },
+};
+
+const configWOMessages: Configurator.Configuration = {
+  owner: {
+    id: PRODUCT_CODE,
+    type: CommonConfigurator.OwnerType.PRODUCT,
+  },
+  configId: CONFIG_ID,
+  productCode: PRODUCT_CODE,
+};
+const configWithMessages: Configurator.Configuration = {
+  owner: {
+    id: PRODUCT_CODE,
+    type: CommonConfigurator.OwnerType.PRODUCT,
+  },
+  configId: CONFIG_ID,
+  productCode: PRODUCT_CODE,
+  errorMessages: ['test error message 1', 'test error message 2'],
+  warningMessages: [
+    'test warning message 1',
+    'test warning message 2',
+    'test warning message 3',
+  ],
+};
+const configWithOnlyOneMessage: Configurator.Configuration = {
+  owner: {
+    id: PRODUCT_CODE,
+    type: CommonConfigurator.OwnerType.PRODUCT,
+  },
+  configId: CONFIG_ID,
+  productCode: PRODUCT_CODE,
+  errorMessages: ['test error message 1'],
+  warningMessages: ['test warning message 1'],
+};
+
+let configuration: Configurator.Configuration;
+
+class MockRoutingService {
+  getRouterState(): Observable<RouterState> {
+    return of(mockRouterState);
+  }
+}
+
+class MockRouter {
+  public events = of('');
+}
+
+class MockConfiguratorCommonsService {
+  getConfiguration(): Observable<Configurator.Configuration> {
+    return of(configuration);
+  }
+}
+
+export class MockIconFontLoaderService {
+  getFlipDirection(): void {}
+}
+
+@Component({
+  selector: 'cx-icon',
+  template: '',
+})
+class MockCxIconComponent {
+  @Input() type: any;
+}
+
+describe('ConfiguratorConflictAndErrorMessagesComponent', () => {
+  let component: ConfiguratorConflictAndErrorMessagesComponent;
+  let fixture: ComponentFixture<ConfiguratorConflictAndErrorMessagesComponent>;
+  let configuratorUtils: CommonConfiguratorUtilsService;
+  let htmlElem: HTMLElement;
+
+  beforeEach(
+    waitForAsync(() => {
+      TestBed.configureTestingModule({
+        imports: [I18nTestingModule, ReactiveFormsModule, NgSelectModule],
+        declarations: [
+          ConfiguratorConflictAndErrorMessagesComponent,
+          MockCxIconComponent,
+        ],
+        providers: [
```

Comment:
This component only depends on `ConfiguratorCommonsService` and `ConfiguratorRouterExtractorService`. So, we only need to mock these 2 services.

Reply:
Picked!

Reply:
Done!